hibike/devices/YogiBear/YogiBear.cpp

```diff
@@ -1,122 +1,324 @@
+      return sizeof(float);
       break;
+    case PID_VEL_KI: 
+    	setVelKI(((float *)data)[0]);
+      return sizeof(float);
+      break;
+    case PID_VEL_KD: 
+    	setVelKD(((float *)data)[0]);
+      return sizeof(float);
+      break;
+    case CURRENT_THRESH: 
+    	setCurrentThreshold(((float *)data)[0]);
+      return sizeof(float);
+      break;
+    case ENC_POS: 
+    	if((float) data[0] == 0) {
+    		zeroEncoder();
+      return sizeof(float);
+    	}
+      break;
+    case ENC_VEL: 
+      break;
+    case MOTOR_CURRENT: 
+      break;
+    default:
+      return 0;
+    }
+
+  /*
+    case VALUE_0:
+      value0 = (uint8_t) read_num_bytes(data, sizeof(value0));
+      return sizeof(value0);
+      break;
+    case VALUE_1:
+      value1 = (uint8_t) read_num_bytes(data, sizeof(value1));
+      return sizeof(value1);
+      break;
+    case VALUE_2:
+      value2 = (uint16_t) read_num_bytes(data, sizeof(value2));
+      return sizeof(value2);
+      break;
+    case VALUE_3:
+      value3 = (uint32_t) read_num_bytes(data, sizeof(value3));
+      return sizeof(value3);
+      break;
+    case VALUE_4:
+      value4 = (uint64_t) read_num_bytes(data, sizeof(value4));
+      return sizeof(value4);
+      break;
+    case VALUE_5:
+      value5 = (uint64_t) read_num_bytes(data, sizeof(value5));
+      return sizeof(value5);
+      break;
+    default:
+      return 0;
   }
-  return ~((uint32_t) 0);
+  */
+  return 0;
 }
 
 
-// you must implement this function. It is called with a buffer and a maximum buffer size.
-// The buffer should be filled with appropriate data for a DataUpdate packer, and the number of bytes
-// added to the buffer should be returned. 
+// You must implement this function.
+// It is called when the device receives a Device Data Update packet.
+// Modifies data_update_buf to contain the parameter value.
+//    param           -   Parameter index
+//    data_update_buf -   buffer to return data in, little-endian
+//    buf_len         -   Maximum length of the buffer
 //
-// You can use the helper function append_buf.
-// append_buf copies the specified amount data into the dst buffer and increments the offset
-uint8_t data_update(uint8_t* data_update_buf, size_t buf_len) {
-  uint8_t offset = 0;
-  return offset;
-}
+//    return          -   sizeof(value) on success; 0 otherwise
 
+uint8_t device_read(uint8_t param, uint8_t* data_update_buf, size_t buf_len) {
+  double reading;
+  float* float_buf;
+  switch (param) 
+  {
+  	case ENABLE:
+  	  data_update_buf[0] = enabled;
+  	  return sizeof(uint8_t);
+  	  break;
+    case COMMAND_STATE: 
+      data_update_buf[0] = driveMode;
+      return sizeof(driveMode);
+      break;
+    case DUTY_CYCLE: 
+      float_buf = (float *) data_update_buf;
+      float_buf[0] = pwmInput;
+      return sizeof(pwmInput);
+      break;
+    case PID_POS_SETPOINT: 
+      // float_buf = (float *) data_update_buf;
+      // float_buf[0] = PIDPos;
```

Comment:
Are these parameters write-only? Either implement them or delete the commented out code